docs/proposals/timeseries-partitioning-in-compactor.md

```diff
@@ -0,0 +1,267 @@
+
+Now the planner knows the resulting compaction will have 8 partitions, it can start planning out which groups of blocks can go into a single compaction group. Given that we need 8 partitions in total, the planner will go through the process above to find out what blocks are necessary. Using the above example again, but we have distinct time intervals, T1, T2, and T3. T1 has 2 partitions, T2 has 4 partitions, and T3 has 8 partitions, and we want to produce T1-T3 blocks
+![Grouping](/images/proposals/timeseries-partitioning-in-compactor-grouping.png)
+```
+Compaction Group 1-8
+T1 - Partition 1-2
+T2 - Partition 1-4
+T3 - Partition 1-8
+
+Compaction Group 2-8
+T1 - Partition 2-2
+T2 - Partition 2-4
+T3 - Partition 2-8
+
+Compaction Group 3-8
+T1 - Partition 1-2
+T2 - Parittion 3-4
+T3 - Partition 3-8
+
+Compaction Group 4-8
+T1 - Partition 2-2
+T2 - Partition 4-4
+T3 - Partition 4-8
+
+Compaction Group 5-8
+T1 - Partition 1-2
+T2 - Partition 1-4
+T3 - Partition 5-8
+
+Compaction Group 6-8
+T1 - Partition 2-2
+T2 - Partition 2-4
+T3 - Partition 6-8
+
+Compaction Group 7-8
+T1 - Partition 1-2
+T2 - Partition 3-4
+T3 - Partition 7-8
+
+Compaction Group 8-8
+T1 - Partition 2-2
+T2 - Partition 4-4
+T3 - Partition 8-8
+```
+
+`T1 - Partition 1-2` is used in multiple compaction groups, and the following section will describe how the compaction avoids duplicate timeseries in the resulting blocks
+
+### Compaction
+
+Now that the planner has produced a compaction plan for the T1-T3 compaction groups, the compactor can start downloading the necessary blocks. Using compaction group 1-8 from above as example.
+![Grouping](/images/proposals/timeseries-partitioning-in-compactor-compact.png)
+T1 - Partition 1-2 was created with hash % 2 == 0, and in order to avoid having duplication information in blocks produced by compaction group 3-8, compaction group 5-8, and compaction group 7-8, we need apply the filter the `%8 == 0` hash, as that’s the hash of the highest partition count.
+
+## Performance
+
+Currently a 400M timeseries takes 12 hours to compact, without taking block download into consideration. If we have a partition count of 2, we can reduce this down to 6 hours, and a partition count of 10 is 3 hours. The scaling is not linear, and I’m still attempting to find out why. The initial result is promising enough to continue though.
+
+## Alternatives Considered
+
+### Dynamic Number of Partition
+
+We can also increase/decrease the number of partition without needing the `multiplier` factor. However, if a tenant is sending highly varying number of timeseries or label size, the index size can be very different, resulting in highly dynamic number of partitions. To perform deduplication, we’ll end up having to download all the sub-blocks, and it can be inefficient as less parallelization can be done, and we will spend more time downloading all the unnecessary blocks.
```

Comment:
Thanks for putting this here. I was wondering why does it have to be this complex.